pkg/amqp/subscriber.go

```diff
@@ -238,6 +234,11 @@ func (s *Subscriber) runSubscriber(
 		}
 	}()
 
+	if err = s.prepareConsume(queueName, exchangeName, logFields); err != nil {
+		s.logger.Error("Failed to prepare consume", err, logFields)
+		return
+	}
```

Comment:
@reva2 The issue here might be that you won't receive this error on the first subscriber run (it's just logged). Perhaps it makes sense to run `s.prepareConsume` in all these cases?

Reply:
Although that might cause duplicated logs so it's also not an elegant solution. 🤔